Tasks/SshV0/ssh2helpers.ts

```diff
@@ -63,23 +63,33 @@ function handleStreamClose(command: string, stdErrWritten: boolean, defer: Q.Def
 }
 
 /**
- * Uses scp2 to copy a file to remote machine
- * @param scriptFile
- * @param scpConfig
+ * Uses sftp to copy a file to remote machine
+ * @param src
+ * @param dest
+ * @param sftpConfig
  * @returns {Promise<string>}
  */
-export function copyScriptToRemoteMachine(scriptFile: string, scpConfig: any): Q.Promise<string> {
+export async function copyScriptToRemoteMachine(src: string, dest: string, sftpConfig: SftpClient.ConnectOptions): Promise<string> {
     const defer = Q.defer<string>();
+    const sftpClient = new SftpClient();
+
+    try {
+        await sftpClient.connect(sftpConfig);
+        await sftpClient.put(src, dest);
+        tl.debug('Copied script file to remote machine at: ' + dest);
+        defer.resolve();
+    } catch (err) {
+        defer.reject(tl.loc('RemoteCopyFailed', err));
+    }
 
-    scp2.scp(scriptFile, scpConfig, (err) => {
-        if (err) {
-            defer.reject(tl.loc('RemoteCopyFailed', err));
-        } else {
-            tl.debug('Copied script file to remote machine at: ' + scpConfig.path);
-            defer.resolve(scpConfig.path);
-        }
-    });
-
+    try {
+        sftpClient.on('error', (err) => {
+            tl.debug('sftpClient: Ignoring error diconnecting: ' + err);
+        }); // ignore logout errors; see: https://github.com/mscdex/node-imap/issues/695
```

Comment:
Nit: I assume it will be better to provide more details in the code comments instead of a link to the GitHub issue.

Reply:
Added some more description here